docs/source/1.0/spec/core/idl.rst

```diff
@@ -190,12 +190,14 @@ The Smithy IDL is defined by the following ABNF:
     enum_shape_statement   :`enum_type_name` `ws` `identifier` [`mixins`] `ws` `enum_shape_members`
     enum_type_name         :"enum" / "intEnum"
     enum_shape_members     :"{" `ws` 1*(`trait_statements` `identifier` `ws`) "}"
-    shape_members          :"{" `ws` *(`shape_member_kvp` `ws`) "}"
-    shape_member_kvp       :`trait_statements` `identifier` `ws` ":" `ws` `shape_id`
+    shape_members          :"{" `ws` *(`trait_statements` `shape_member` `ws`) "}"
+    shape_member           :`shape_member_kvp` / `shape_member_elided`
+    shape_member_kvp       :`identifier` `ws` ":" `ws` `shape_id`
+    shape_member_elided    :"$" `identifier`
     list_statement :"list" `ws` `identifier` [`mixins`] `ws` `shape_members`
     set_statement :"set" `ws` `identifier` [`mixins`] `ws` `shape_members`
     map_statement :"map" `ws` `identifier` [`mixins`] `ws` `shape_members`
-    structure_statement     :"structure" `ws` `identifier` [`mixins`] `ws` `shape_members`
+    structure_statement     :"structure" `ws` `identifier` ["from" `shape_id`] [`mixins`] `ws` `shape_members`
```

Comment:
we've been saying "for" in mockups, and I can't help but think "for" is better than "from".